MapleServer2/Packets/StatPacket.cs

```diff
@@ -17,27 +17,23 @@ public static Packet SetStats(IFieldObject<Player> player)
             return pWriter;
         }
 
-        public static Packet UpdateMobStats(IFieldObject<Mob> mob)
+        public static Packet UpdateMobStats(IFieldObject<Mob> mob, SkillCast skillCast)
         {
             PacketWriter pWriter = PacketWriter.Of(SendOp.STAT);
             pWriter.WriteInt(mob.ObjectId);
             pWriter.WriteByte();
             pWriter.WriteByte(1);
             pWriter.WriteByte(4); // value
-            // Stats 
-            // Damage should be update through this packet
-            for (int i = 0; i < 3; i++)
-            {
-                pWriter.WriteLong(mob.Value.Stats.Hp[i]);
-            }
+            pWriter.WriteLong(mob.Value.Stats.Hp.Total);
+            pWriter.WriteLong(mob.Value.Stats.Hp.Min);
+            pWriter.WriteLong(mob.Value.UpdateStats(mob.Value, skillCast.GetDamage()));
```

Comment:
Again, why are we changing mob stats in the middle of the packet? Can this be done outside the packet method? Also, why do we need to pass in `mob.Value` when we already have access to it